Docs cleanup and typo fixes (#1557)
nautobot/nautobot: nautobot/docs/rest-api/overview.md

### Non-Breaking Changes

Non-breaking (forward- and backward-compatible) REST API changes may be introduced in major or minor Nautobot releases. Since these changes are non-breaking, they will _not_ correspond to the introduction of a new API version, but will be added seamlessly to the existing API version, and so will immediately be available to existing REST API clients. Examples would include:

* Addition of new fields in GET responses

> * Understanding of new, _optional_ fields in POST/PUT/PATCH requests

Reviewer: Perhaps:
```suggestion
* Added support for new, _optional_ fields in POST/PUT/PATCH requests
```

Most API endpoints support an optional "brief" format, which returns only a minimal representation of each object in the response. This is useful when you need only a list of available objects without any related data, such as when populating a drop-down list in a form. As an example, the default (complete) format of an IP address looks like this:

```json
GET /api/ipam/prefixes/7d2d24ac-4737-4fc1-a850-b30366618f3d/

> {

Author: Sure if you prefer.  I personally thought the syntax highlighting from the `json` tag made it easier to read.
The other sections further down the page, such as in the pagination section, have json formatting already in there.

Thoughts?

> {

Reviewer: I was suggesting applying `no-highlight` only to the first line as a separate block, since it's not actually part of the JSON, then keeping `json` for the actual response body. Sorry that was unclear!

> {

Author: Ah gotcha.  Thank you!
